### PR TITLE
Override TokenRefreshView to update JWT_AUTH_COOKIE cookie

### PR DESCRIPTION
**Pull request by lemirep**
Add a dedicated view that subclasses django-rest-framework-simplejwt's
TokenRefreshView to update the JWT_AUTH_COOKIE on successful refreshes.

Fixes: #97

### CODE REVIEW

**Comment by lemirep** [comment]
Please let me know if there's a macro or comment I can put to please coverage bot on lines that can't  be reached in the testing environment

**Comment by iMerica** [comment]
Some jwt/cookie enhancements were recently added. Can you rebase your branch with upstream?

**Comment by lemirep** [comment]
Rebased against latest changes

**Comment by alichass** [comment]
Hi sorry I have a question; if the purpose of putting the jwt in the httponly cookie is to prevent it from potentially being accessed from the browser storage (local storage), doesn't storing the refresh token to then use to refresh the cookie cause the same venerability?

I think a http-only JWT refresh cookie might solve this problem, but im wondering if that then becomes a whole deal with csrf

**Comment by lemirep** [comment]
I'm new to this stuff so I could be wrong but from what I've been reading on https://hasura.io/blog/best-practices-of-using-jwt-with-graphql/#refresh_token_persistance, @alichass I think you're right. The access token is supposed to be short lived so it makes more sense if the refresh token, which is valid for longer and therefore more sensitive, is stored securely in a cookie.
Storing the access token in a cookie makes things more convenient as you don't have to pass it with every request. @iMerica what do you think?

**Comment by alichass** [comment]
Hi @lemirep . Yes normally the access token is short lived so that if it is compromised (lets say stolen from your local web storage), the expiry time kinda deals with the threat; however, storing the access token in an HTTPOnly cookie is meant to mitigate this problem as the token no longer is accessible through js (assuming you dont also store the tokens returned on login in your local storage) . See the comment here: https://github.com/jazzband/dj-rest-auth/issues/97

This means that you dont really need to worry about the JWT token being compromised in the same way, and so having a short lived expiry time on the access token doesn't really do anything (seeing as the token is invalidated on logout - i think).

tldr; using cookie jwt with a long enough expirey time means you dont need to worry about the whole refresh token flow, and makes it effectively useless - if not a security issue if saved in local storage

If for instance you want your api to be be used **both** with JWT header based auth (tokens are stored in local storage on the client) **and** with JWT cookie based auth for an SPA (or SPAs ), then yes you'd probably want to set the access to a short expiry time - and thats when this PR seems super useful. 

If for instance you want your api to be used with **only** JWT cookie based auth for an SPA, and JWT header based auth for some app with a low risk of exposing the tokens (eg a backend client) then you dont have to worry and you can simply have the JWT access token expiry set to something large.

**Comment by lemirep** [comment]
Seemed interesting to have but not getting any feedback about what needs changing to get this integrated. Dropping this patch.
